pygount/command.py

```diff
@@ -327,35 +328,29 @@ def execute(self):
         )
         source_paths_and_groups_to_analyze = list(source_scanner.source_paths())
         duplicate_pool = pygount.analysis.DuplicatePool() if not self.has_duplicates else None
-        if self.output == "STDOUT":
-            target_file = sys.stdout
-            has_target_file_to_close = False
-        else:
-            target_file = open(self.output, "w", encoding="utf-8", newline="")
-            has_target_file_to_close = True
+        writer_class = _OUTPUT_FORMAT_TO_WRITER_CLASS_MAP[self.output_format]
 
-        try:
-            writer_class = _OUTPUT_FORMAT_TO_WRITER_CLASS_MAP[self.output_format]
-
-            if issubclass(writer_class, pygount.write.SummaryWriter):
-                source_paths_and_groups_to_analyze = track(source_paths_and_groups_to_analyze)
-            with writer_class(target_file) as writer:
-                for source_path, group in source_paths_and_groups_to_analyze:
-                    statistics = pygount.analysis.SourceAnalysis.from_file(
-                        source_path,
-                        group,
-                        self.default_encoding,
-                        self.fallback_encoding,
-                        generated_regexes=self._generated_regexs,
-                        duplicate_pool=duplicate_pool,
-                    )
-                    writer.add(statistics)
-        finally:
-            if has_target_file_to_close:
+        with Progress(transient=True) as progress:
```

Comment:
Progress is not down there with the other context managers because the code right below it references sys.stdout.

Thing is: rich makes progress appear below all printed text by replacing sys.stdout with its wrapper during the Progress.__enter__. However, if we make a reference to sys.stdout before __enter__, the progress bar display will be broken.

But because it's a whole another level of indentation, I decided to refactor the code below which is why we have so many changes. It is **almost** equivalent. Except for the fact that we don't raise our own OSError.

Reply:
Makes sense.

Didn't know about `contextlib.nullcontext()`, learned something new. 🙂 

I'm not so sure about removing rid of the `OSError`. After the merge I'm going to this to ensure in case of write errors we still get a meaningful error message.